src/cmd-buildextend-installer

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+#
+# An operation that creates an ISO image for installing CoreOS
+
+import os
+import sys
+import json
+import yaml
+import shutil
+import argparse
+
+sys.path.insert(0, '/usr/lib/coreos-assembler')
+from cmdlib import run_verbose, write_json, sha256sum_file
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--build", help="Build ID")
+args = parser.parse_args()
+
+# default to latest build if not specified
+if not args.build:
+    with open('builds/builds.json') as f:
+        j = json.load(f)
+        args.build = j['builds'][0]
+
+print(f"Targeting build: {args.build}")
+
+builddir = os.path.join('builds', args.build)
+buildmeta_path = os.path.join(builddir, 'meta.json')
+with open(buildmeta_path) as f:
+    buildmeta = json.load(f)
+
+base_name = buildmeta['name']
+img_prefix = f'{base_name}-{args.build}'
+iso_name = f'{img_prefix}.iso'
+
+tmpdir = 'tmp/buildpost-installer'
+if os.path.isdir(tmpdir):
+    shutil.rmtree(tmpdir)
+
+tmpisoroot = os.path.join(tmpdir, 'isolinux')
+tmpostreedir = os.path.join(tmpdir, 'ostree')
+
+os.mkdir(tmpdir)
+os.mkdir(tmpisoroot)
+
+def generate_iso():
+
+    tmpisofile = os.path.join(tmpdir, iso_name)
+
+    # Extract the kernel/initrd from the ostree commit related to this build
+    buildmeta_commit = buildmeta['ostree-commit']
+    run_verbose(['/usr/bin/ostree', '--repo=./repo', 'checkout',
+                 '--user-mode', f"{buildmeta_commit}", 
+                 '--subpath=/usr/lib/ostree-boot', tmpostreedir])
+    run_verbose('/usr/bin/mv {} {}'.format(
```

Comment:
While this is OK here passing filenames into the shell is a general source of problems (and for e.g. web apps, security vulnerabilities).  

Related to the above, what you want is the first non-empty dir in `/usr/lib/modules/` and then it's just `vmlinuz` and `initramfs.img`.  See the libostree source related to this.